### ISSUE TITLE
Property accessor hack for SQLAlchemy columns

### ISSUE THREAD

**Issue by layday** [microsoft/pyright#444]
**Describe the bug**
The SQLAlchemy stubs from [Dropbox](https://github.com/dropbox/sqlalchemy-stubs) use descriptors for `Column` attributes returning the underlying type (in this case, a string) for objects of the enclosing class [[2]](https://github.com/dropbox/sqlalchemy-stubs/blob/d6cf578/sqlalchemy-stubs/sql/schema.pyi#L71-L134):

```py
class Column(SchemaItem, ColumnClause[_T]):
    ...
    @overload
    def __get__(self, instance: None, owner: Any) -> Column[_T]: ...
    @overload
    def __get__(self, instance: object, owner: Any) -> _T: ...
```

This is not being picked up by pyright.  Compare with mypy:

<img width="546" alt="Screenshot 2019-12-21 at 03 50 58" src="https://user-images.githubusercontent.com/31134424/71301317-5a045c80-23a6-11ea-8ce3-6e9134333ef8.png">

**To Reproduce**
Add the SQLAlchemy stubs in `typingsPath` and type check the snippet below.

**Expected behavior**
`Foo().bar` should be of type `str`.

**Screenshots or Code**

```py
from sqlalchemy import Column, String


class Foo:
    bar = Column(String)


reveal_type(Foo.bar)
reveal_type(Foo().bar)

```

**VS Code extension or command-line**
extension, 1.1.11

**Comment by erictraut** [comment]
Thanks for the bug report. Pyright wasn't properly handling overloads for the `__get__` method. This will be fixed in the next version of Pyright.

**Comment by erictraut** [comment]
This is now fixed in version 1.1.12, which I just published.

**Comment by hassanselim0** [comment]
@erictraut This is very similar to my issue the issue I'm facing (overloads on `__get__` to achieve accurate ORM types) but with Django Stubs instead of SQLAlchemy. In my case Pylance (or Pyright) says the type is `Any` instead of `str`.
Here is the relevant part in the Django Stubs repo: https://github.com/typeddjango/django-stubs/blob/master/django-stubs/db/models/fields/__init__.pyi#L107

Either this issue has regressed, or there is a bug in the django stubs package. Honestly python generics and overloads are still relatively new to me and I struggle with deciphering them, so any help is appreciated.

**Comment by erictraut** [comment]
I did find a bug in Pyright. it was not properly handling the `__get__` methods with generic types. I don't know if this will completely fix the issue at hand.

You are correct that django-stubs relies on a mypy plugin for some of its behavior. Mypy plugins are very specific to mypy. They don't work with other type checkers like pyright.

**Comment by hassanselim0** [comment]
Yeah I'm almost certain that django-stubs was made in such a way that orm field type would only work on mypy, that `_pyi_private_get_type` attr is not a standard and I found code inside a mypy plugin folder that parses it.

But it's good that you found that bug, even if it wont solve my problem.
